### PR TITLE
Improve XI API and client

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

A suite of various improvements around the XI.

## Proposal

- [x] fix missing xi routers DB rollbacks
- [x] make XI client more developer-friendly
- [x] use the XI client in the seeding script

### CODE REVIEW

File: src/api/core/warren/xi/client.py [hunk 1]
```diff
@@ -190,22 +215,39 @@ async def delete(self, **kwargs) -> None:
...
+        db_relations = await self.read(
+            source=data.source_id, target=data.target_id, kind=data.kind.value
+        )
+
+        # Create
+        if db_relations is None or len(db_relations) == 0:
+            return await self.create(data)
+
+        # Update
+        db_relation = db_relations.pop()
```
Comment (wilbrdt) [inline src/api/core/warren/xi/client.py:229]: Could we have multiple relations objects with the same source, target and kind?
Should we check if there are multiple relations objects before updating the last read?

File: src/api/core/warren/xi/routers/experiences.py [hunk 2]
```diff
@@ -64,6 +66,10 @@ async def read_experiences(  # noqa: PLR0913
         statement = statement.where(Experience.aggregation_level == aggregation_level)
     if iri:
         statement = statement.where(Experience.iri == iri)
+    if technical_datatypes:
+        statement = statement.filter(
+            Experience.technical_datatypes.comparator.contains(technical_datatypes)  # type: ignore[union-attr]
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/api/core/warren/xi/routers/experiences.py:71]: Yup, that's it :point_right: https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.JSON.Comparator

File: src/api/core/warren/xi/client.py (hunk 1 above)
Comment (jmaupetit) [reply src/api/core/warren/xi/client.py:229]: No we can't: we have a unique constraint on the `("source_id", "target_id", "kind")` triplet.

Reference: https://github.com/openfun/warren/blob/main/src/api/core/warren/xi/schema.py#L49

Comment (wilbrdt) [reply src/api/core/warren/xi/client.py:229]: Great! No problem here then

**Review by wilbrdt — APPROVED** [review]

**Pull request merged**